```no-error-kafka``` CI fail is not related to this PR.
```
Error: eckstyle] [ERROR] /home/runner/work/checkstyle/checkstyle/.ci-temp/kafka/tools/src/main/java/org/apache/kafka/tools/ConfigCommand.java:109: Javadoc comment at column 39 has parse error. Details: token recognition error at: '|' while parsing brokers [UnusedImports]
```

@romani good to go?

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/YieldHandler.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/YieldHandler.java
@@ -42,6 +42,18 @@ public class YieldHandler extends AbstractExpressionHandler {
     }
 
     @Override
+    public IndentLevel getSuggestedChildIndent(AbstractExpressionHandler child) {
+        final IndentLevel result;
+        if (TokenUtil.areOnSameLine(getMainAst(), child.getMainAst())) {
+            result = getIndent();
+        }
+        else {
+            result = super.getSuggestedChildIndent(child);
+        }
+        return result;
+    }
+
+    @Override
     public void checkIndentation() {
         checkYield();
         final DetailAST expression = getMainAst().getFirstChild();

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java
@@ -3859,6 +3859,34 @@ public class IndentationCheckTest extends AbstractModuleTestSupport {
     }
 
     @Test
+    public void testIndentationYieldWithNestedSwitchExpression() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(IndentationCheck.class);
+        checkConfig.addProperty("tabWidth", "4");
+        checkConfig.addProperty("basicOffset", "4");
+        checkConfig.addProperty("braceAdjustment", "0");
+        checkConfig.addProperty("caseIndent", "4");
+        checkConfig.addProperty("lineWrappingIndentation", "4");
+
+        final String fileName = getPath("InputIndentationYieldSwitchExpression.java");
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+        verifyWarns(checkConfig, fileName, expected);
+    }
+
+    @Test
+    public void testIndentationYieldWithWrappedSwitchExpression() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(IndentationCheck.class);
+        checkConfig.addProperty("tabWidth", "4");
+        checkConfig.addProperty("basicOffset", "4");
+        checkConfig.addProperty("braceAdjustment", "0");
+        checkConfig.addProperty("caseIndent", "4");
+        checkConfig.addProperty("lineWrappingIndentation", "4");
+
+        final String fileName = getPath("InputIndentationYieldSwitchExpressionWrapped.java");
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+        verifyWarns(checkConfig, fileName, expected);
+    }
+
+    @Test
     public void testInputIndentationLambdaChild() throws Exception {
         final DefaultConfiguration checkConfig = createModuleConfig(IndentationCheck.class);
         checkConfig.addProperty("tabWidth", "4");

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationYieldSwitchExpression.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationYieldSwitchExpression.java
@@ -1,0 +1,27 @@
+
+package com.puppycrawl.tools.checkstyle.checks.indentation.indentation;      //indent:0 exp:0
+
+/* Config:                                                                   //indent:0 exp:0
+ * This test-input is intended to be checked using following configuration:  //indent:1 exp:1
+ * tabWidth = 4                                                              //indent:1 exp:1
+ * basicOffset = 4                                                           //indent:1 exp:1
+ * braceAdjustment = 0                                                       //indent:1 exp:1
+ * caseIndent = 4                                                            //indent:1 exp:1
+ * lineWrappingIndentation = 4                                               //indent:1 exp:1
+ */                                                                          //indent:1 exp:1
+
+public class InputIndentationYieldSwitchExpression {                         //indent:0 exp:0
+    void foo() {                                                             //indent:4 exp:4
+        int i = 0;                                                           //indent:8 exp:8
+        int j = switch (i) {                                                 //indent:8 exp:8
+            case 0 -> {                                                      //indent:12 exp:12
+                yield switch (i) {                                           //indent:16 exp:16
+                    case 0 -> 0;                                             //indent:20 exp:20
+                    default -> 1;                                            //indent:20 exp:20
+                };                                                           //indent:16 exp:16
+            }                                                                //indent:12 exp:12
+            default -> 1;                                                    //indent:12 exp:12
+        };                                                                   //indent:8 exp:8
+    }                                                                        //indent:4 exp:4
+}                                                                            //indent:0 exp:0
+

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationYieldSwitchExpressionWrapped.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationYieldSwitchExpressionWrapped.java
@@ -1,0 +1,27 @@
+
+package com.puppycrawl.tools.checkstyle.checks.indentation.indentation;      //indent:0 exp:0
+
+/* Config:                                                                   //indent:0 exp:0
+ * This test-input is intended to be checked using following configuration:  //indent:1 exp:1
+ * tabWidth = 4                                                              //indent:1 exp:1
+ * basicOffset = 4                                                           //indent:1 exp:1
+ * braceAdjustment = 0                                                       //indent:1 exp:1
+ * caseIndent = 4                                                            //indent:1 exp:1
+ * lineWrappingIndentation = 4                                               //indent:1 exp:1
+ */                                                                          //indent:1 exp:1
+
+public class InputIndentationYieldSwitchExpressionWrapped {                  //indent:0 exp:0
+    int bar(int i) {                                                         //indent:4 exp:4
+        return switch (i) {                                                  //indent:8 exp:8
+            case 0 -> {                                                      //indent:12 exp:12
+                yield                                                        //indent:16 exp:16
+                    switch (i) {                                             //indent:20 exp:20
+                        case 0 -> 0;                                         //indent:24 exp:24
+                        default -> 1;                                        //indent:24 exp:24
+                    };                                                       //indent:20 exp:20
+            }                                                                //indent:12 exp:12
+            default -> 1;                                                    //indent:12 exp:12
+        };                                                                   //indent:8 exp:8
+    }                                                                        //indent:4 exp:4
+}                                                                            //indent:0 exp:0
+

